test/masq.test.js

```diff
@@ -28,204 +30,246 @@ let profile = {
+})
 
-  sw.on('close', () => {
-    done()
+test('should be kicked if challenge does not match', async () => {
+  expect.assertions(2)
+
+  const pr = new Promise((resolve, reject) => {
+    const { channel } = masq.requestMasqAccess()
+
+    // simulating masq app
+    const hub = signalhub(channel, [HUB_URL])
+    const sw = swarm(hub, { wrtc })
+    sw.on('peer', (peer, id) => {
+      expect(sw.peers).toHaveLength(1)
+      peer.send(JSON.stringify({
+        msg: 'sendProfilesKey',
+        challenge: 'challengemismatch'
+      }))
+    })
+
+    sw.on('disconnect', (peer, id) => {
+      expect(peer).toBeDefined()
+      sw.close()
+    })
+
+    sw.on('close', () => {
+      resolve()
+    })
   })
 
-  masq.requestMasqAccess()
+  await pr
+  await masq.requestMasqAccessDone()
 })
 
-test('should replicate masq-profiles', async (done) => {
+test('should receive message replicationProfilesStarted', async () => {
   expect.assertions(1)
-  masq = new Masq()
-  await masq.init()
 
-  const challenge = masq.challenge
-  const channel = masq.channel
+  const pr = new Promise((resolve, reject) => {
+    const { channel, challenge } = masq.requestMasqAccess()
+    const key = 'c4b60362325d27ad3c04db158fa68fe6fde00387467708ab3a0be79c811b3825'
 
-  /***
-   * A link must be created (and sent to masq) containing :
-   *  type: 'syncProfiles',
-   *  channel: channel,
-   *  challenge: challenge
-   */
-  const hub = signalhub(channel, [HUB_URL])
-  const sw = swarm(hub, { wrtc })
-  const key = 'c4b60362325d27ad3c04db158fa68fe6fde00387467708ab3a0be79c811b3825'
+    // simulating masq app
+    const hub = signalhub(channel, [HUB_URL])
+    const sw = swarm(hub, { wrtc })
 
-  sw.on('peer', (peer, id) => {
-    peer.on('data', data => {
-      const json = JSON.parse(data)
+    sw.on('peer', (peer, id) => {
+      peer.on('data', data => {
+        const json = JSON.parse(data)
+        expect(json.msg).toBe('replicationProfilesStarted')
+        sw.close()
+      })
 
-      expect(json.msg).toBe('replicationProfilesStarted')
-      sw.close()
+      peer.send(JSON.stringify({
+        msg: 'sendProfilesKey',
+        challenge: challenge,
+        key: key
+      }))
     })
 
-    peer.send(JSON.stringify({
-      msg: 'sendProfilesKey',
-      challenge: challenge,
-      key: key
-    }))
-  })
-
-  sw.on('close', () => {
-    hub.close()
-  })
+    sw.on('close', () => {
+      resolve()
+    })
 
-  sw.on('disconnect', (peer, id) => {
-    sw.close()
-    done()
+    sw.on('disconnect', (peer, id) => {
+      sw.close()
+    })
   })
 
-  masq.requestMasqAccess()
+  await pr
+  await masq.requestMasqAccessDone()
 })
 
+async function _initProfilesReplication () {
+  const pr = new Promise((resolve, reject) => {
+    const { channel, challenge } = masq.requestMasqAccess()
+    const key = 'c4b60362325d27ad3c04db158fa68fe6fde00387467708ab3a0be79c811b3825'
+
+    // simulating masq app
+    const hub = signalhub(channel, [HUB_URL])
```

Comment:
I think we should move this type of mock in its own file, will see later

Reply:
Good idea ! to be added to the backlog :+1: